src/neural/onnx/adapters.cc

```diff
@@ -57,4 +58,22 @@ std::string FloatOnnxWeightsAdapter::GetRawData() const {
   }
 }
 
-}  // namespace lczero
\ No newline at end of file
+pblczero::TensorProto::DataType Float16OnnxWeightsAdapter::GetDataType() const {
+  return pblczero::TensorProto::FLOAT16;
+}
+
+std::string Float16OnnxWeightsAdapter::GetRawData() const {
+  std::vector<uint16_t> fp16(weights_.size());
+  std::transform(weights_.begin(), weights_.end(), fp16.begin(), FP32toFP16);
+  if (order_.empty()) {
```

Comment:
Totally optional, but as the code repeats now, maybe add something like this function to transpose.h (or even here in adapters.cc in anonymous namespace):

```cpp
template <class T>
std::string TransposeAndReturnRaw(const std::vector<int>& dims, std::vector<int> order,
                     const std::vector<T>& from) {
  if (order.empty()) {
    return {reinterpret_cast<const char*>(from.data()),
            reinterpret_cast<const char*>(from.data() + weights_.size())};
  } else {
    std::vector<T> dst(weights_.size());
    TransposeTensor(dims, order, weights, &dst[0]);
    return {reinterpret_cast<const char*>(dst.data()),
            reinterpret_cast<const char*>(dst.data() + dst.size())};
  }
}
```